routes/api_v1_0.py

```diff
@@ -400,28 +795,150 @@ def timestamps():
 def topo_v1_0(shaded_relief: str, projection: str, zoom: int, x: int, y: int):
-    hull_shade = shaded_relief == 'true'
-    return routes.routes_impl.topo_impl(projection, zoom, x, y, hull_shade)
+    """
+        Generates topographical tiles
+    """
+
+    if shaded_relief == "true":
+        bShaded_relief = True
+    else:
+        bShaded_relief = False
+
+    shape_file_dir = current_app.config['SHAPE_FILE_DIR']
+
+    if zoom > 7:
+        return send_file(shape_file_dir + "/blank.png")
+
+    cache_dir = current_app.config['CACHE_DIR']
+    f = os.path.join(cache_dir, request.path[1:])
+
+    if os.path.isfile(f):
+        return send_file(f, mimetype='image/png', cache_timeout=MAX_CACHE)
+    else:
+        bytesIOBuff = plotting.tile.topo(projection, x, y, zoom, bShaded_relief)
+
+        return _cache_and_send_img(bytesIOBuff, f)
 
 
 @bp_v1_0.route('/api/v1.0/tiles/bath/<string:projection>/<int:zoom>/<int:x>/<int:y>.png')
 def bathymetry_v1_0(projection: str, zoom: int, x: int, y: int):
-    return routes.routes_impl.bathymetry_impl(projection, zoom, x, y)
+    """
+       Generates bathymetry tiles
+    """
+
+    shape_file_dir = current_app.config['SHAPE_FILE_DIR']
+
+    if zoom > 7:
+        return send_file(shape_file_dir + "/blank.png")
+
+    cache_dir = current_app.config['CACHE_DIR']
+    f = os.path.join(cache_dir, request.path[1:])
+
+    if os.path.isfile(f):
+        return send_file(f, mimetype='image/png', cache_timeout=MAX_CACHE)
+    else:
+        img = plotting.tile.bathymetry(projection, x, y, zoom, {})
+        return _cache_and_send_img(img, f)
 
 
 @bp_v1_0.route('/api/v1.0/mbt/<string:projection>/<string:tiletype>/<int:zoom>/<int:x>/<int:y>')
 def mbt(projection: str, tiletype: str, zoom: int, x: int, y: int):
-    return routes.routes_impl.mbt_impl(projection, tiletype, zoom, x, y)
+    """
+         Serves mbt files
+    """
+    cache_dir = current_app.config['CACHE_DIR']
+    shape_file_dir = current_app.config['SHAPE_FILE_DIR']
+    requestf = str(os.path.join(cache_dir, request.path[1:]))
+    basedir = requestf.rsplit("/", 1)[0]
+
+    # Send blank tile if conditions aren't met
+    if (zoom < 7) or (projection != "EPSG:3857"):
+        return send_file(shape_file_dir + "/blank.mbt")
+
+    # Send file if cached or select data in SQLite file
+    if os.path.isfile(requestf):
+        return send_file(requestf)
+    else:
+        y = (2**zoom-1) - y
+        connection = sqlite3.connect(
+            shape_file_dir + "/{}.mbtiles".format(tiletype))
+        selector = connection.cursor()
+        sqlite = "SELECT tile_data FROM tiles WHERE zoom_level = {} AND tile_column = {} AND tile_row = {}".format(
+            zoom, x, y)
+        selector.execute(sqlite)
+        tile = selector.fetchone()
+        if tile == None:
+            return send_file(shape_file_dir + "/blank.mbt")
+
+        # Write tile to cache and send file
+        if not os.path.isdir(basedir):
```

Comment:
Would `if os.path.isdir(basedir) is None:` make sense here? Not saying it should be changed...just another way of writing this to consider...